source/features/clean-conversation-filters.tsx

```diff
@@ -42,44 +43,33 @@ function getCount(element: HTMLElement): number {
 	return Number(element.textContent.trim());
 }
 
-async function hideMilestones(): Promise<void> {
-	const milestones = await elementReady('[data-selected-links^="repo_milestones"] .Counter');
-	if (getCount(milestones!) === 0) {
-		(await elementReady('[data-hotkey="m"]'))!.parentElement!.remove();
+const MILESTONE_SELECTORS = ['#milestones-select-menu', '[data-testid="action-bar-item-milestones"]'];
+const PROJECT_SELECTORS = ['#project-select-menu', '[data-testid="action-bar-item-projects"]'];
+
+async function hide(container: HTMLElement): Promise<void> {
+	const milestones = $('[data-selected-links^="repo_milestones"] .Counter');
+	if (milestones && getCount(milestones) === 0) {
+		$(MILESTONE_SELECTORS, container)?.remove();
+	}
+
+	if (await hasAnyProjects.get()) {
+		return;
 	}
-}
 
-async function hideProjects(): Promise<void> {
-	const projectsDropdown = await elementReady('[data-hotkey="p"]');
-	projectsDropdown?.parentElement!.remove();
+	const projectsDropdown = $(PROJECT_SELECTORS, container);
+	projectsDropdown?.remove();
```

Comment:
Why `?`

When I encounter `| undefined` on Refined GitHub I have to make decide between:

- I know it exists in this view
- I know it might not exist in this view

In both cases I have to know.

In the first case, either I use `!` or in this case `expectElement` so that when it's missing I get a proper error.

In the latter, I leave a comment explaining in what situation it might be missing. 

When I don't know for sure, I just assume it exists and then wait for the errors to appear or be reported.

Errors help me see when selectors break; `?.` instead just hides errors

Reply:
Oh, I get it. We should get an error if it doesn't exist.